cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/YamlMapper.java

```diff
@@ -112,13 +105,19 @@ public static String resolveYamlFile(String yamlFilePath) throws IOException {
      * @param resolvedYamlContent the yaml content to be interpreted as a bean object
      * @param beanType the desired type of the bean
      * @return the resolved content of the yaml file in the yaml format
-     * @throws ParserException if the configuration file content cannot be parsed as yaml
-     * @throws ScannerException if the provided yaml content cannot be interpreted as yaml
-     * @throws ConstructorException if the resolved config cannot be interpreted as the given bean type
+     * @throws IllegalArgumentException if the configuration file content cannot be parsed as yaml
+     * @throws YamlParsingException if the resolved config cannot be interpreted as the given bean type
      */
     public static <B extends Bean> B interpretBean(String resolvedYamlContent, Class<B> beanType) {
         Yaml yamlProcessor = new Yaml();
-        return yamlProcessor.loadAs(resolvedYamlContent, beanType);
+
+        try {
+            return yamlProcessor.loadAs(resolvedYamlContent, beanType);
+        } catch (ConstructorException constructorException) {
+            throw new YamlParsingException(constructorException);
+        } catch (MarkedYAMLException markedYamlException) {
+            throw new IllegalArgumentException("Unable to parse the resolved yaml content", markedYamlException);
```

Comment:
Please comment why you opted to throw this exception here. In other occasions, you wrap the exception differently.

Reply:
Good point